JeMPI_Apps/JeMPI_Linker/src/main/java/org/jembi/jempi/linker/BackEnd.java

```diff
@@ -466,4 +485,39 @@ public record EventLinkPatientToGidSyncRsp(
          LinkInfo linkInfo) implements EventResponse {
    }
 
+   public static ArrayList<Notification.MatchData> getCandidates(final PatientRecord patientRecord) throws RuntimeException {
+
+      try {
+         List<GoldenRecord> candidateGoldenRecords =
+               libMPI.getCandidates(patientRecord.demographicData(), AppConfig.BACK_END_DETERMINISTIC);
+         List<WorkCandidate> allCandidateScores =
+               candidateGoldenRecords.parallelStream()
+                                     .unordered()
+                                     .map(candidate -> new WorkCandidate(candidate,
```

Comment:
maybe we could chain a ".filter()" function after this to get candidates with scores within threshold and avoid having a "for" loop afterwards. Does it sound right/better ?